src/axl.c

```diff
@@ -62,142 +99,243 @@ API Functions
 
 /* Read configuration from non-AXL-specific file
   Also, start up vendor specific services */
-int AXL_Init (char* conf_file) {
-    axl_next_handle_UID = 0;
-    axl_flush_async_file_lists = kvtree_new();
+int AXL_Init (const char* state_file)
+{
+    int rc = AXL_SUCCESS;
 
-    char* axl_cntl_dir = NULL;
-    axl_read_config(&axl_cntl_dir);
+    /* TODO: set these by config file */
+    axl_file_buf_size = (size_t) 1048576;
 
-    // TODO: what is the flush file for?
-    char* axl_flush_file_name = "/axl_flush.info";
-    axl_flush_file = malloc(strlen(axl_cntl_dir) + strlen(axl_flush_file_name));
-    strcpy(axl_flush_file, axl_cntl_dir);
-    strcat(axl_flush_file, axl_flush_file_name);
+    /* make a copy of the path to file to record our state */
+    if (state_file != NULL) {
+        axl_flush_file = strdup(state_file);
+    }
 
-    if (axl_cntl_dir != NULL) {
-      free(axl_cntl_dir);
+    axl_next_handle_UID = 0;
+    axl_file_lists = kvtree_new();
+
+    /* initialize values from state file if we have one */
+    if (axl_flush_file != NULL) {
+        /* initialize axl_file_lists from file if we have one */
+        kvtree_read_file(axl_flush_file, axl_file_lists);
+
+        /* initialize handle id using highest id in file */
+        kvtree* ids = kvtree_get(axl_file_lists, AXL_KEY_HANDLE_UID);
+        kvtree_sort_int(ids, KVTREE_SORT_DESCENDING);
+        kvtree_elem* elem = kvtree_elem_first(ids);
+        if (elem != NULL) {
+            int id = kvtree_elem_key_int(elem);
+            axl_next_handle_UID = id;
+        }
     }
 
 #ifdef HAVE_DAEMON
-    return axl_flush_async_init_daemon();
+    char axl_async_daemon_path[] = "axld";
+    char axl_async_daemon_file[] = "/dev/shm/axld";
+    if (axl_async_init_daemon(axl_async_daemon_path, axl_async_daemon_file) != AXL_SUCCESS) {
+        rc = AXL_FAILURE;
+    }
 #endif
 #ifdef HAVE_BBAPI
-    return axl_flush_async_init_bbapi();
```

Comment:
I did this in case something earlier in that function sets rc=FAILURE before it calls axl_flush_async_init_bbapi.  We just need to be careful that if rc has already been set to FAILURE that we don't overwrite that with SUCCESS due to a call to init_bbapi (which might succeed).  The other option would be something like:

if (rc == AXL_SUCCESS) {
  rc = axl_flush_async_init_bbapi();
}